Core/Core/Assignments/AssignmentList/View/AssignmentGroupView.swift

```diff
@@ -20,28 +20,53 @@ import SwiftUI
 
 public struct AssignmentGroupView: View {
 
+    @State private var isExpanded: Bool = true
     @ObservedObject private var viewModel: AssignmentGroupViewModel
 
     public init(viewModel: AssignmentGroupViewModel) {
         self.viewModel = viewModel
     }
 
     public var body: some View {
-        Section(header: ListSectionHeaderOld { Text(viewModel.name) }) {
-            ForEach(viewModel.assignments, id: \.id) { assignment in
-                let assignmentCellViewModel = AssignmentCellViewModel(assignment: assignment, courseColor: viewModel.courseColor)
-                VStack(spacing: 0) {
-                    AssignmentCellView(viewModel: assignmentCellViewModel)
-
-                    if viewModel.assignments.last != assignment {
-                        Divider()
+        Section(header: ListSectionHeaderOld(backgroundColor: .backgroundLightest) { headerView() }) {
+            if isExpanded {
+                ForEach(viewModel.assignments, id: \.id) { assignment in
+                    let assignmentCellViewModel = AssignmentCellViewModel(assignment: assignment, courseColor: viewModel.courseColor)
+                    VStack(spacing: 0) {
+                        AssignmentCellView(viewModel: assignmentCellViewModel)
+
+                        if viewModel.assignments.last != assignment {
+                            InstUI.Divider()
+                        }
                     }
                 }
             }
         }
         .listRowInsets(EdgeInsets())
         .listRowSeparator(.hidden)
     }
+
+    @ViewBuilder
+    private func headerView() -> some View {
```

Comment:
I think you don't this method to be tagged with `@ViewBuilder` since it only returns single view.